pallets/author-inherent/src/lib.rs

```diff
@@ -81,51 +79,54 @@ pub mod pallet {
 		CannotBeAuthor,
 	}
 
-
 	/// Author of current block.
 	#[pallet::storage]
 	pub type Author<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(_: T::BlockNumber) -> Weight {
+			// Start by clearing out the previous block's author
 			<Author<T>>::kill();
+
+			// Now extract the author from the digest
+			let digest = <frame_system::Pallet<T>>::digest();
+
+			let pre_runtime_digests = digest.logs.iter().filter_map(|d| d.as_pre_runtime());
+			Self::find_author(pre_runtime_digests).map(|author_account|{
```

Comment:
If find_author return None, we should panic here to make the bloc invalid.

Reply:
Good catch on this one.

As we discussed in private, I've added a new empty-payload inherent to kick off the author validation as a work-aroud until https://github.com/paritytech/substrate/pull/10128 lands.